pac4j-jwt/src/main/java/org/pac4j/jwt/credentials/authenticator/JwtAuthenticator.java

```diff
@@ -160,14 +188,38 @@ private static void createJwtProfile(final TokenCredentials credentials, final S
      * @since 1.8.2
      */
     public String getSigningSecret() {
-        return signingSecret;
+        return new String(key.getEncoded(), Charset.forName("UTF-8"));
     }
 
     /**
      * @since 1.8.2
      */
     public void setSigningSecret(final String signingSecret) {
-        this.signingSecret = signingSecret;
+    	X509Certificate cert = X509CertUtils.parse(signingSecret);
+    	if (cert == null) {
+    		this.key = new SecretKeySpec(signingSecret.getBytes(Charset.forName("UTF-8")), "AES");
+    	} else {
+    		this.key = cert.getPublicKey();
+    	}
+    }
+    
+    /**
+     * @throws NoSuchAlgorithmException 
+     * @throws InvalidKeySpecException 
+     * @since 1.8.2
```

Comment:
I was expected the `@since 1.8.9` javadoc as it's a new method and we are at version 1.8.9